hackathon_site/dashboard/frontend/src/slices/event/teamAdminSlice.ts

```diff
@@ -34,47 +35,51 @@ export const initialState = teamAdminAdapter.getInitialState(extraState);
 
 // Thunks
 
-export const getAllTeams = createAsyncThunk<
+export const getTeamsWithSearchThunk = createAsyncThunk<
     APIListResponse<Team>,
-    void,
+    string | undefined,
     { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
->(`${teamAdminReducerName}/getAllTeams`, async (_, { dispatch, rejectWithValue }) => {
-    try {
-        const response = await get<APIListResponse<Team>>("/api/event/teams/", {
-            limit: NUM_TEAM_LIMIT,
-        });
-        return response.data;
-    } catch (e: any) {
-        dispatch(
-            displaySnackbar({
-                message: `Failed to fetch team data: Error ${e.response.status}`,
-                options: { variant: "error" },
-            })
-        );
-        return rejectWithValue({
-            status: e.response.status,
-            message: e.response.data,
-        });
+>(
+    `${teamAdminReducerName}/getTeamsWithSearchThunk`,
+    async (search, { dispatch, rejectWithValue }) => {
+        try {
+            const response = await get<APIListResponse<Team>>("/api/event/teams/", {
+                limit: NUM_TEAM_LIMIT,
+                search,
```

Comment:
i think we should make this property conditional (only add it if search is not undefined), something like this:
```
{
    limit: <>,
    (!!search && { search })
}
```